pages/verification/index.tsx

```diff
@@ -64,11 +72,12 @@ export default function Verification(props: PageProps) {
                 }, {
                     onSuccess: (data) => {
                         if (data) {
-                            setIsLoading(false);
-                            router.push('/message-sent');
+                            if(replaceRoute) router.replace('/message-sent');
+                            else router.push('/message-sent');
                         }
                         localStorage.removeItem(MessageEnum.tempMessage);
                         localStorage.removeItem(MessageEnum.targetUser);
+                        setIsLoading(false);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes here introduce a conditional route replacement based on the `replaceRoute` flag. This is a good approach to handle different routing needs. However, the removal of local storage items and setting `isLoading` to false should be done regardless of whether data exists or not. This ensures that the state is cleaned up properly even if the request fails.

Consider moving these lines outside the `if (data)` block.
```diff
                     onSuccess: (data) => {
                         if (data) {
                             if(replaceRoute) router.replace('/message-sent');
                             else router.push('/message-sent');
                         }
-                          localStorage.removeItem(MessageEnum.tempMessage);
-                          localStorage.removeItem(MessageEnum.targetUser);
-                         setIsLoading(false);
                     },
+                    localStorage.removeItem(MessageEnum.tempMessage);
+                    localStorage.removeItem(MessageEnum.targetUser);
+                    setIsLoading(false);
                     onError: (err) => {
                         setIsLoading(false);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->